modules/library/jdbc/src/main/java/org/geotools/jdbc/SessionCommandsListener.java

```diff
@@ -159,36 +159,37 @@ Expression expandEviromentVariables(String sql) {
                 inEnvVariable = true;
                 i++;
             } else if (curr == '}') {
-                if (!inEnvVariable)
-                    throw new IllegalArgumentException(
-                            "Already found a ${ sequence before the one at " + (i + 1));
-
-                if (sb.length() == 0)
-                    throw new IllegalArgumentException(
-                            "Invalid empty environment variable reference ${} at " + (i - 1));
+                if (inEnvVariable) {
```

Comment:
The fix looks good, looking at it now it seems that doing a``if (curr == '}' && inEnvVariable)`` would make it better (one less nesting).